### ISSUE TITLE
Unable to compile with go v1.21

### ISSUE THREAD

**Issue by gr0l0rg** [prometheus/prometheus#13128]
### What did you do?

The version `v0.47.2` is imported by the prometheus otel collector

```
../go/pkg/mod/github.com/prometheus/prometheus@v0.47.2/model/labels/labels.go:365:23: type func(a Label, b Label) bool of func(a, b Label) bool {…} does not match inferred type func(a Label, b Label) int for func(a E, b E) int
```

### What did you expect to see?

successful compilation

### What did you see instead? Under which circumstances?

Code does not compile

### Prometheus version

```text
v0.47.2
```

**Comment by mmorel-35** [comment]
This seems to be fixed since 2.48.0.rc.0

https://github.com/prometheus/prometheus/blob/551fa8378cce4219e2ed3e591bc26d33ce6e25cf/model/labels/labels.go#L366

**Comment by gr0l0rg** [comment]
@mmorel-35 Thank you! I will open an issue in the otel-contrib repo so that the maintainers can upgrade the dependency when it's released.
